test/unit/codegen/codegen_helper.cpp

```diff
@@ -168,3 +185,33 @@ SCENARIO("unusual / failing mod files", "[codegen][var_order]") {
         }
     }
 }
+
+SCENARIO("Testing that passes are correct") {
+    GIVEN("DERIVATIVE should generate two for loop") {
+        std::string nmodl_text = R"(
+            NEURON {
+                SUFFIX hh
+                RANGE minf, mtau
+            }
+            STATE {
+                m
+            }
+            ASSIGNED {
+                v (mV)
+                minf
+                mtau (ms)
+            }
+            BREAKPOINT {
+                SOLVE states METHOD cnexp
+            }
+            DERIVATIVE states {
+                m = (minf-m)/mtau
+            }
+        )";
+
+        THEN("CHECK IT") {
+            auto result = run_inline_visitor_helper(nmodl_text);
+            REQUIRE(result.size() == 2);
```

Comment:
we should check the contents of individual node as well. For example, something like:
```
auto main_loop = to_nmodl(result[0]);
// here check main_loop against expected
// same for remainder loop
```